scan/context.go

```diff
@@ -88,35 +87,40 @@ func (s *Scanner) getContextFromGitURL(gitURL string) (contextDir string, err er
 }
 
 func (s *Scanner) getContextFromURL(remoteURL string) (err error) {
-	response, err := getWithStatusError(remoteURL)
-	if err != nil {
-		return errors.Wrapf(err, "unable to download remote context from %s", remoteURL)
-	}
+	attempt := 0
+	for attempt < maxRetries {
+		var response *http.Response
+		response, err = getWithStatusError(remoteURL)
+		if err != nil {
+			return errors.Wrap(err, "unable to download remote context")
+		}
 
-	// TODO: revamp streaming, for now just pipe to buf and discard it.
-	var buf bytes.Buffer
```

Comment:
Can just use `io.Discard`.

Reply:
Is the behavior (what we eventually log) the same as before after the pr changes to `io.Discard`?

Reply:
The progress bar is not printed before this PR, because the `buf` is not used.
```
var buf bytes.Buffer
progressOutput := streamformatter.NewProgressOutput(&buf)
```
The behavior remains the same if change to 
```
progressOutput := streamformatter.NewProgressOutput(io.Discard)
```